loki/transformations/transform_loop.py

```diff
@@ -294,140 +294,145 @@ def loop_fusion(routine):
-                        if is_new_bound:
-                            # Remove any lower bounds made redundant by bound:
-                            upper_bounds = [b for b, d in zip(upper_bounds, diff)
-                                            if not (is_constant(d) and symbolic_op(d, op.gt, 0))]
-                            upper_bounds += [bound]
-
-                if len(lower_bounds) == 1:
-                    lower_bounds = lower_bounds[0]
-                else:
-                    fct_symbol = sym.ProcedureSymbol('min', scope=routine)
-                    lower_bounds = sym.InlineCall(fct_symbol, parameters=as_tuple(lower_bounds))
-
-                if len(upper_bounds) == 1:
-                    upper_bounds = upper_bounds[0]
-                else:
-                    fct_symbol = sym.ProcedureSymbol('max', scope=routine)
-                    upper_bounds = sym.InlineCall(fct_symbol, parameters=as_tuple(upper_bounds))
-
-                fusion_ranges += [sym.LoopRange((lower_bounds, upper_bounds))]
-
-        # Align loop ranges and collect bodies
-        fusion_bodies = []
-        fusion_variables = loop_variables[0]
-        for idx, (variables, ranges, bodies, p) in enumerate(
-                zip(loop_variables, loop_ranges, loop_bodies, iteration_spaces)):
-            # TODO: This throws away anything that is not in the inner-most loop body.
-            body = flatten([Comment(f'! Loki loop-fusion - body {idx} begin'),
-                            bodies[-1],
-                            Comment(f'! Loki loop-fusion - body {idx} end')])
-
-            # Replace loop variables if necessary
-            var_map = {}
-            for loop_variable, fusion_variable in zip(variables, fusion_variables):
-                if loop_variable != fusion_variable:
-                    var_map.update({var: fusion_variable for var in FindVariables().visit(body)
-                                    if var.name.lower() == loop_variable.name})
-            if var_map:
-                body = SubstituteExpressions(var_map).visit(body)
-
-            # Wrap in conditional if loop bounds are different
-            conditions = []
-            for loop_range, fusion_range, variable in zip(ranges, fusion_ranges, fusion_variables):
-                if symbolic_op(loop_range.start, op.ne, fusion_range.start):
-                    conditions += [sym.Comparison(variable, '>=', loop_range.start)]
-                if symbolic_op(loop_range.stop, op.ne, fusion_range.stop):
-                    conditions += [sym.Comparison(variable, '<=', loop_range.stop)]
-            if conditions:
-                if len(conditions) == 1:
-                    condition = conditions[0]
-                else:
-                    condition = sym.LogicalAnd(as_tuple(conditions))
-                body = Conditional(condition=condition, body=as_tuple(body), else_body=())
+        if not fusion_groups:
```

Comment:
[no action] This entire chunk looks like it's just been unindented. Is that right, or are there any other changes in here? Tests look fine, so I'm not worried, just curious.

Reply:
I am a little bit confused to be honest.
It used to be that loop fusion (and early exit) was outside of `pragmas_attached`. Currently, it is within `pragmas_attached`.
If I put the loop fusion outside the `pragmas_attached` (through removing indentation) I get spurious Loki pragmas in the transformed code, but like it is now I don't get any `!$loki loop-fusion group ...` in the transformed source?!